components/mediators/datamapper/org.wso2.carbon.mediator.datamapper/src/main/java/org/wso2/carbon/mediator/datamapper/DataMapperMediator.java

```diff
@@ -282,32 +283,38 @@ private void transform(MessageContext synCtx, String configKey, String inSchemaK
                 JsonUtil.newJsonPayload(axis2MessageContext, outputResult, true, true);
             }
         } catch (ReaderException | InterruptedException | XMLStreamException | SchemaException
-                | IOException | WriterException e) {
+                | IOException | JSException | WriterException e ) {
             handleException("DataMapper mediator : mapping failed", e, synCtx);
         }
     }
 
     private InputStream getInputStream(MessageContext context, String inputType, String inputStartElement) {
         InputStream inputStream = null;
-        switch (InputOutputDataType.fromString(inputType)) {
-        case XML:
-        case CSV:
-            if ("soapenv:Envelope".equals(inputStartElement)) {
+        try {
+            switch (InputOutputDataType.fromString(inputType)) {
+            case XML:
+            case CSV:
+                if ("soapenv:Envelope".equals(inputStartElement)) {
+                    inputStream = new ByteArrayInputStream(
+                            context.getEnvelope().toString().getBytes(StandardCharsets.UTF_8));
+                } else {
+                    inputStream = new ByteArrayInputStream(context.getEnvelope().getBody().getFirstElement().toString()
+                            .getBytes(StandardCharsets.UTF_8));
+                }
+                break;
+            case JSON:
+                org.apache.axis2.context.MessageContext a2mc = ((Axis2MessageContext) context).getAxis2MessageContext();
+                if (JsonUtil.hasAJsonPayload(a2mc)) {
+                    inputStream = JsonUtil.getJsonPayload(a2mc);
+                }
+                break;
+            default:
                 inputStream = new ByteArrayInputStream(
                         context.getEnvelope().toString().getBytes(StandardCharsets.UTF_8));
-            } else {
-                inputStream = new ByteArrayInputStream(
-                        context.getEnvelope().getBody().getFirstElement().toString().getBytes(StandardCharsets.UTF_8));
-            }
-            break;
-        case JSON:
-            org.apache.axis2.context.MessageContext a2mc = ((Axis2MessageContext) context).getAxis2MessageContext();
-            if (JsonUtil.hasAJsonPayload(a2mc)) {
-                inputStream = JsonUtil.getJsonPayload(a2mc);
             }
-            break;
-        default:
-            inputStream = new ByteArrayInputStream(context.getEnvelope().toString().getBytes(StandardCharsets.UTF_8));
+        } catch (Exception e) {
```

Comment:
Is there any particular reason for catching generic Exception here? We can catch the specific exception and handle.